commit a07033a6c15d0104521862c2709341cdf7e91952 (from f8199b2cc15dbc30ee9a8e857a25116447ca60bf)
Merge: f8199b2 00456b0
Author: Scott Lahteine <thinkyhead@users.noreply.github.com>
Date:   Mon Oct 10 13:27:52 2016 -0500

    Merge pull request #4974 from esenapaj/Follow-up-the-PR-#4955
    
    Fix and follow-up the PR #4955 (PINS_DEBUGGING and M43: Read pin states), etc

diff --git a/Marlin/Marlin_main.cpp b/Marlin/Marlin_main.cpp
index 2e46ffb..14843d4 100755
--- a/Marlin/Marlin_main.cpp
+++ b/Marlin/Marlin_main.cpp
@@ -4700,7 +4700,7 @@ inline void gcode_M42() {
           pin_state[pin - first_pin] = digitalRead(pin);
       }
 
-      #if ENABLED(EMERGENCY_PARSER)
+      #if ENABLED(EMERGENCY_PARSER) && DISABLED(ULTIPANEL)
         wait_for_user = true;
       #endif
 
@@ -4718,7 +4718,7 @@ inline void gcode_M42() {
           }
         }
 
-        #if ENABLED(EMERGENCY_PARSER)
+        #if ENABLED(EMERGENCY_PARSER) && DISABLED(ULTIPANEL)
           if (!wait_for_user) break;
         #endif
 

